src/gui/owncloudgui.cpp

```diff
@@ -51,61 +53,46 @@ namespace {
-            // this is needed to ensure a consistent state in the config file upon unexpected terminations of the client
-            // (for instance, when running from a debugger and stopping the process from there)
-            AccountManager::instance()->save(true);
-            auto folderMan = FolderMan::instance();
-
-            // saves a bit of duplicate code
-            auto addFolder = [folderMan, accountStatePtr, syncMode](const QString &localFolder, const QString &remotePath, const QUrl &webDavUrl, const QString &displayName = {}) {
-                folderMan->addFolderFromWizard(accountStatePtr, localFolder, remotePath, webDavUrl, displayName, syncMode);
-            };
-
-            auto finalize = [accountStatePtr] {
-                accountStatePtr->checkConnectivity();
-                FolderMan::instance()->scheduleAllFolders();
-            };
-
-            if (accountStatePtr->account()->capabilities().spacesSupport().enabled) {
-                auto *drive = new OCC::GraphApi::Drives(accountStatePtr->account());
-
-                QObject::connect(drive, &OCC::GraphApi::Drives::finishedSignal, [accountStatePtr, localFolder, drive, addFolder, finalize] {
-                    if (drive->parseError().error == QJsonParseError::NoError) {
-                        const auto &drives = drive->drives();
-                        if (!drives.isEmpty()) {
-                            const QDir localDir(localFolder);
-                            localDir.mkdir(".");
-                            for (const auto &d : drives) {
-                                const QDir driveLocalFolder = localDir.filePath(d.getName());
-                                driveLocalFolder.mkdir(".");
-                                addFolder(driveLocalFolder.absolutePath(), {}, QUrl::fromEncoded(d.getRoot().getWebDavUrl().toUtf8()), d.getName());
-                            }
-                            finalize();
-                        }
-                    }
-                });
+    auto folderMan = FolderMan::instance();
 
-                drive->start();
+    // saves a bit of duplicate code
+    auto addFolder = [folderMan, accountStatePtr, useVfs](const QString &localFolder, const QString &remotePath, const QUrl &webDavUrl, const QString &displayName = {}) {
+        folderMan->addFolderFromWizard(accountStatePtr, localFolder, remotePath, webDavUrl, displayName, useVfs);
+    };
 
-                return;
-            } else {
-                addFolder(localFolder, Theme::instance()->defaultServerFolder(), accountStatePtr->account()->davUrl());
-                finalize();
+    auto finalize = [accountStatePtr] {
+        accountStatePtr->checkConnectivity();
+        FolderMan::instance()->scheduleAllFolders();
+    };
+
+    if (accountStatePtr->account()->capabilities().spacesSupport().enabled) {
+        auto *drive = new OCC::GraphApi::Drives(accountStatePtr->account());
```

Comment:
Obviously out of scope of this PR, but something that was a bit confusing to me: GraphApi::Drives is actually a Job... most jobs here also are named BlaJob, but this one is not. Maybe that would be helpful.